src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1102,18 +1164,34 @@ private void onDecodeUnsubscribe(
             final int packetId = unsubscribe.packetId();
 
             MqttTopicFW topic;
+            boolean error = false;
```

Comment:
Retain the `int` error code instead of a `boolean`, initialized to `NO_ERROR` then becomes `PROTOCOL_ERROR`.